env/work/srw_python/srwl_uti_smp.py

```diff
@@ -0,0 +1,208 @@
+        self.limit_value = d['limit_value']
+
+    def get_image_from_data(self):
+        try:
+            import numpy as np
+        except:
+            raise ImportError('The specified npy file cannot be read since {}'.format(self._import_error_msg.format('numpy')))
+        data = np.load(self.file_path)
+        self.limit_value = 255
+        self.data = (data - data.min()) / (data.max() - data.min()) * self.limit_value
+        try:
+            from PIL import Image
+            self.raw_image = Image.fromarray(self.data.astype(np.uint8))
+            self.processed_image = self.raw_image
+        except:
+            print('The processed image will not be saved since {}'.format(self._import_error_msg.format('pillow')))
+            self.is_save_images = False
+
+    def read_sample(self):
+        if self.input_type == 'image':
+            self.get_data_from_image()
+        elif self.input_type == 'npy':
+            self.get_image_from_data()
+        else:
+            raise NotImplementedError(
+                'Processing of the "{}" input type is not implemented yet.'.format(self.input_type))
+        self.nx = self.data.shape[0]
+        self.ny = self.data.shape[1]
+
+    def save_images(self):
+        # self.raw_image.save(os.path.join(self.save_dir, self.raw_image_name))
+        self.processed_image.save(os.path.join(self.save_dir, self.processed_image_name))
+
+    def show_images(self):
+        self.raw_image.show()
+        self.processed_image.show()
+
+    def _add_prefix(self, prefix, name, image_format):
+        output_name = '{}_{}'.format(prefix, name) if prefix else name
+        return '{}.{}'.format(output_name, image_format)
+
+    def _check_files(self):
+        if not os.path.isfile(self.file_path):
+            raise ValueError('Provided file "{}" does not exist.'.format(self.file_path))
+
+    def _check_input_type(self, file_path):
+        self.possible_extensions = {
+            'image': ['tif', 'tiff', 'png', 'bmp', 'gif', 'jpg', 'jpeg'],
+            'npy': ['npy'],
+        }
+        extension = os.path.splitext(file_path)[1][1:].lower()
+        for k in self.possible_extensions.keys():
+            for e in self.possible_extensions[k]:
+                if extension == e:
+                    return k
+        all_extensions = [x for x_list in self.possible_extensions.values() for x in x_list]
+        all_extensions += [x.upper() for x in all_extensions]
+        raise ValueError('Incorrect extension: {}. Possible values: {}.'.format(extension, ', '.join(all_extensions)))
+
+
+# ********************** Create transmission element from the data from an image file:
+def srwl_opt_setup_transm_from_file(file_path, resolution, thickness, delta, atten_len,
+                                    arTr=None, extTr=0, fx=1e+23, fy=1e+23,
+                                    xc=0, yc=0, ne=1, e_start=0, e_fin=0,
+                                    is_save_images=True, prefix=''):
```

Comment:
All the arguments for the transmission object added here; extTr set to zero by default and will be provided as an input field in Sirepo later.